common/djangoapps/track/contexts.py

```diff
@@ -12,26 +12,27 @@
 log = logging.getLogger(__name__)
 
 
-def course_context_from_url(url):
+def course_context_from_url(url, course_id_string=None):
     """
-    Extracts the course_context from the given `url` and passes it on to
-    `course_context_from_course_id()`.
+    If course_id_string string is not present, extracts it from the given `url`. Either way, then passes
+    it on to `course_context_from_course_id()`.
     """
     url = url or ''
-
-    match = COURSE_REGEX.match(url)
     course_id = None
-    if match:
-        course_id_string = match.group('course_id')
-        try:
-            course_id = CourseKey.from_string(course_id_string)
-        except InvalidKeyError:
-            log.warning(
-                'unable to parse course_id "{course_id}"'.format(
-                    course_id=course_id_string
-                ),
-                exc_info=True
-            )
+
+    if course_id_string is None:
```

Comment:
Our source for data is going to be messy, and different parts of the codebase treat `None` and blank strings ambiguously. Can you please make this a little broader and make it fall back to the old behavior for blank string values as well?

So something like:

```python
if course_id_string:
    # The parsing of the explicit field passed in
else:
    # Magic parsing from the URL
```

Reply:
done in upcoming push